### PR TITLE
Blog App Milestone 2: Feat/processing data model

### PR DESCRIPTION
**Pull request by andres-condezo**
Rails

---
### Hi there :wave:
Thanks for taking the time to review my project.

--- 
### Changes

In this milestone I:

- Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models. 
- Create the following methods:
  - Users
    - A method that returns the 3 most recent posts for a given user.
  - Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
  - Comments
    - A method that updates the comments counter for a post.
  - Likes
    - A method that updates the likes counter for a post.
- Create the current user method.
---
### Get Started

Clone the repository :
```
git clone https://github.com/andres-condezo/Blog-app.git
cd Blog-app
git checkout feat/processing_data_model
```

### Any comment or suggestion will be thankfully received.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :user_id, dependent: :destroy
+  has_many :comments, foreign_key: :user_id, dependent: :destroy
+  has_many :likes, foreign_key: :user_id, dependent: :destroy
+
```
Comment (Kingstalux) [inline app/models/user.rb]: - [x] Great job so far 👍 kindly change the foreign_key from `user_id` to `author_id` and also kindly change all `user` to `author` so as to strictly follow the [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) diagram 👍 
- [x] Kindly apply this change to all your models 💯

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @andres-condezo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

175083599-879bf673-fb93-406f-939c-de7e586bf6d9

## Highlights 👏 
✅ Associations have been well implemented 🔥 
✅ All models and methods are present 👍 
✅ Good commit history 👌 
✅ Professional README file 💯 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (andres-condezo) [reply app/models/user.rb]: Hi @Kingstalux, thank you for your feedback, I'll work on it, in this moment.

**Review by t-yanick — APPROVED** [review]
Hi @, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean and organized code

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**